### PR TITLE
Add Kinesis ListShards permission

### PR DESCRIPTION
**Pull request by ektravel**
### Description

This PR specifies that the IAM policy must include the Kinesis `ListShards` permission when using the Kinesis indexing service. 

This PR has:
- [ x ] been self-reviewed.
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Review by techdocsmith — APPROVED** [review]
LGTM

**Comment by kfaraz** [comment]
@ektravel , I think these changes have been added in #12383 when the corresponding feature flag was added.
Do we still need the changes in this PR?

**Comment by abhishekagarwal87** [comment]
Since we are changing the docs here, we should list all the IAM permissions needed by the kinesis extension. Right now, it reads like `ListShards` is the only permission needed. There is also a flag to switch between old and new behavior. So we should mention that too. Different permissions are needed based on the value of flag being used.

**Comment by AmatyaAvadhanula** [comment]
The following permissions (ListStreams for all resources, and Get* for GetShardIterator, GetRecords etc for streams of interest) need to be present for ingestion:
[
        {
            "Effect": "Allow",
            "Action": [
                "kinesis:ListStreams"
            ],
            "Resource": [
                "*"
            ]
        },
        {
            "Effect": "Allow",
            "Action": [
                "kinesis:Get*",
            ],
            "Resource": [
                <ARN for shards to be ingested>
            ]
        }
]

If `useListShards` is set to true, please ensure that `ListShards` is present for streams of interest as well:
        {
            "Effect": "Allow",
            "Action": [
                "kinesis:ListShards",
            ],
            "Resource": [
                <ARN for shards to be ingested>
            ]
        }

Otherwise, please ensure that `DescribeStream` is present for those streams:
        {
            "Effect": "Allow",
            "Action": [
                "kinesis:DescribeStream",
            ],
            "Resource": [
                <ARN for shards to be ingested>
            ]
        }

File: docs/development/extensions-core/kinesis-ingestion.md [hunk 1]
```diff
@@ -340,18 +340,63 @@ For Concise bitmaps:
...
+To authenticate with AWS, you must provide your AWS access key and AWS secret key via `runtime.properties`, for example:
 ```
 -Ddruid.kinesis.accessKey=123 -Ddruid.kinesis.secretKey=456
 ```
+
 The AWS access key ID and secret access key are used for Kinesis API requests. If this is not provided, the service will
 look for credentials set in environment variables, via [Web Identity Token](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_oidc.html), in the default profile configuration file, and from the EC2 instance
 profile provider (in this order).
 
+To ingest data from Kinesis, ensure that the policy attached to the IAM role contains the necessary permissions:
```
[the remark this answers is not in the document]
Comment (abhishekagarwal87) [reply docs/development/extensions-core/kinesis-ingestion.md:357]: @AmatyaAvadhanula - do you have any comments on the resource section in my examples?

[the remark this answers is not in the document]
Comment (abhishekagarwal87) [reply docs/development/extensions-core/kinesis-ingestion.md:357]: Done. @AmatyaAvadhanula.

[the remark this answers is not in the document]
Comment (ektravel) [reply docs/development/extensions-core/kinesis-ingestion.md]: @abhishekagarwal87 @AmatyaAvadhanula I've adopted your suggestions. Please review at your convenience.

**Review by abhishekagarwal87 — APPROVED** [review]

**Comment by abhishekagarwal87** [comment]
Thank you @ektravel. can you fix merge conflicts? We can merge it once the conflicts are resolved.

**Pull request merged**